### PR TITLE
Add Saas App property to DCR

### PR DESCRIPTION
**Pull request by Krishanx92**
### Proposed changes in this pull request

Add Saas App property to DCR

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/main/resources/api.identity.oauth.dcr.endpoint.yaml [hunk 1]
```diff
@@ -300,6 +300,8 @@ definitions:
           type: string
       ext_param_sp_template:
         type: string
+      ext_saas_app:
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/main/resources/api.identity.oauth.dcr.endpoint.yaml]: Can we set the default value here to false?

File: components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/bean/ApplicationRegistrationRequest.java [hunk 2]
```diff
@@ -36,6 +36,7 @@
     private String consumerKey = null;
     private String consumerSecret = null;
     private String spTemplateName = null;
+    private Boolean saasApp = null;
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/bean/ApplicationRegistrationRequest.java]: If we set the default value for this to 'false' then we will not need to check for null when consuming this variable.

File: components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/gen/java/org/wso2/carbon/identity/oauth2/dcr/endpoint/dto/RegistrationRequestDTO.java [hunk 3]
```diff
@@ -181,6 +182,15 @@ public void setBackchannelLogoutSessionRequired(boolean backchannelLogoutSession
     this.backchannelLogoutSessionRequired = backchannelLogoutSessionRequired;
   }
 
+  @ApiModelProperty
+  @JsonProperty("ext_saas_app")
+  public Boolean getExtSaasApp() {
+    return extSaasApp;
+  }
```
Comment (aaujayasena) [inline components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/gen/java/org/wso2/carbon/identity/oauth2/dcr/endpoint/dto/RegistrationRequestDTO.java]: Please leave a blank line between two method

File: components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/main/resources/api.identity.oauth.dcr.endpoint.yaml (hunk 1 above)
Comment (Krishanx92) [reply components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/main/resources/api.identity.oauth.dcr.endpoint.yaml]: Fixed

File: components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/bean/ApplicationRegistrationRequest.java (hunk 2 above)
Comment (Krishanx92) [reply components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/bean/ApplicationRegistrationRequest.java]: Fixed

File: components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/gen/java/org/wso2/carbon/identity/oauth2/dcr/endpoint/dto/RegistrationRequestDTO.java (hunk 3 above)
Comment (Krishanx92) [reply components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/gen/java/org/wso2/carbon/identity/oauth2/dcr/endpoint/dto/RegistrationRequestDTO.java]: Fixed

File: components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java [hunk 4]
```diff
@@ -317,7 +319,7 @@ private Application buildResponse(OAuthConsumerAppDTO createdApp) {
     private void updateServiceProviderWithOAuthAppDetails(ServiceProvider serviceProvider,
                                                           OAuthConsumerAppDTO createdApp,
                                                           String applicationOwner,
-                                                          String tenantDomain) throws DCRMException {
+                                                          String tenantDomain, Boolean saasApp) throws DCRMException {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java]: Let's use the primitive boolean type here

File: components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/gen/java/org/wso2/carbon/identity/oauth2/dcr/endpoint/dto/RegistrationRequestDTO.java [hunk 5]
```diff
@@ -28,6 +28,7 @@
   private List<String> responseTypes = new ArrayList<>();
   private String tokenType = null;
   private String spTemplateName = null;
+  private boolean extSaasApp = false;
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/gen/java/org/wso2/carbon/identity/oauth2/dcr/endpoint/dto/RegistrationRequestDTO.java]: Lets name the variable isSaasApp

**Comment by mefarazath** [comment]
@Krishanx92 Shall we have a test case for setting SaaS property via DCR?
You can add a test case to https://github.com/wso2/product-is/blob/8ab6d7c52f4eeac96031cb395ef1b6ccf6323624/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java#L53

**Comment by Krishanx92** [comment]
> @Krishanx92 Shall we have a test case for setting SaaS property via DCR?
> You can add a test case to https://github.com/wso2/product-is/blob/8ab6d7c52f4eeac96031cb395ef1b6ccf6323624/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java#L53

Integration test PR: https://github.com/wso2/product-is/pull/7384

**Pull request closed without merging**